lib/saluki-components/src/destinations/datadog_status_flare/mod.rs

```diff
@@ -0,0 +1,172 @@
+#[allow(unused)]
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+#[allow(unused)]
+use datadog_protos::agent::{
+    GetFlareFilesRequest, GetFlareFilesResponse, GetStatusDetailsRequest, GetStatusDetailsResponse, RemoteAgent,
+};
+use memory_accounting::{MemoryBounds, MemoryBoundsBuilder};
+use saluki_config::GenericConfiguration;
+use saluki_core::components::{
+    destinations::{Destination, DestinationBuilder, DestinationContext},
+    ComponentContext,
+};
+use saluki_env::helpers::remote_agent::RemoteAgentClient;
+use saluki_error::GenericError;
+use saluki_event::DataType;
+use serde::Deserialize;
+use tracing::debug;
+
+const DEFAULT_ID: &str = "agent-data-plane";
+const DEFAULT_DISPLAY_NAME: &str = "Agent Data Plane";
+const DEFAULT_API_ENDPOINT: &str = "localhost:13580";
+
+/// TODO
+/// wrapped client inside an `Option` to fix Deserialization problems
+#[derive(Deserialize)]
+pub struct DatadogStatusFlareConfiguration {
+    #[serde(default = "default_id", rename = "remote_agent_id")]
+    id: String,
+
+    #[serde(default = "default_display_name", rename = "remote_agent_display_name")]
+    display_name: String,
```

Comment:
Similar to the above, we should always derive the display name from `saluki_metadata::get_app_details()` using the "full name" in its unadulterated form.